bin/logos.pl

```diff
@@ -236,7 +236,7 @@ package main;
 			$currentClass = $currentGroup->addClassNamed($1);
 			$classes{$currentClass->name}++;
 			patchHere(undef);
-		} elsif($line =~ /\G%subclass\s+([\$_\w]+)\s*:\s*([\$_\w]+)\s*(\<\s*(.*?)\s*\>)?/gc) {
+		} elsif($line =~ /\G%subclass\s+([^\s]+)\s*:\s*([^\s]+)\s*(\<\s*(.*?)\s*\>)?/gc) {
```

Comment:
The problem is likely here: matching "not-space" followed by a `:` will match `x:y:z:a:b:c` because it's greedy.

You could experiment with replacing the `+` with `+?` in each of the changed sections.